src/layout/actions/layout.rs

```diff
@@ -924,27 +924,32 @@ impl LayoutTree {
             let parent_ix = self.tree.parent_of(child_ix)
                 .expect("Node had no parent");
             let children = self.tree.children_of(parent_ix);
-            let index = children.iter().position(|&node_ix| node_ix == child_ix);
-            let index_str = index.map(|num| (num + 1).to_string());
 
             // I can't do it inside the match ahead because it borrows
             // self.tree, and I need self.tree to get the children's titles
-            let titles = children.iter()
-                .filter(|&&node_ix| {
-                    !self.tree[node_ix].floating()
-                })
-                .map(|&node_ix| {
+            let mut titles = Vec::<String>::with_capacity(children.len());
+            let mut index = None;
+            let mut filtered_idx = -1;
+            for &node_ix in children.iter() {
+                if self.tree[node_ix].floating() { continue; }
+
+                titles.push(
                     match self.tree[node_ix] {
                         Container::Container { ref borders, .. } |
                         Container::View { ref borders, .. } =>
-                            borders.as_ref().map( |b| b.title() )
+                            borders.as_ref().map(|b| b.title())
                             .unwrap_or("").into(),
                         ref child => {
                             error!("Container child is {:#?}", child);
                             panic!("Container childr is not a Container nor a View")
                         }
                     }
-                }).collect();
+                );
+
+                filtered_idx += 1;
+                if node_ix == child_ix { index = Some(filtered_idx as usize); }
```

Comment:
It looks like you're just getting the index in the list. Again, I don't like this iterative solution so this probably won't matter but for the future you can use `enumerate` to have the index passed to you (like it works in Python, for example).